src/createClient.spec.ts

```diff
@@ -351,6 +351,49 @@ describe('useQuery', () => {
 
     expect(value.response).toBe(response)
   })
+
+  testInEffectScope('immediate false, acts like null parameters', async () => {
+    const input = ref<boolean | null>(null)
+    const action = vi.fn((_value: boolean) => Symbol())
+    const { useQuery } = createClient()
+
+    const query = useQuery(action, () => {
+      if(input.value === null) {
+        return null
+      }
+
+      return [input.value]
+    }, { immediate: false })
+
+    await vi.runAllTimersAsync()
+
+    expect(query.response).toBeUndefined()
+    expect(query.executed).toBe(false)
+    expect(query.executing).toBe(false)
+    expect(action).not.toHaveBeenCalled()
+
+    input.value = true
+
+    await vi.runAllTimersAsync()
+
+    expect(query.response).toBeUndefined()
+    expect(query.executed).toBe(false)
+    expect(query.executing).toBe(false)
+    expect(action).not.toHaveBeenCalled()
```

Comment:
Id add to these tests (or add a new test) that calling execute overrides this and starts the query.